test/gopls_tests.vim

```diff
@@ -12,30 +12,91 @@ call LspAddServer(lspServers)
 echomsg systemlist($'{lspServers[0].path} version')
 
 # Test for :LspGotoDefinition, :LspGotoDeclaration, etc.
+# This test also tests that multiple locations will be
+# shown in a list or popup
 def g:Test_LspGoto()
   :silent! edit Xtest.go
   sleep 200m
 
   var lines =<< trim END
     package main
 
-    func foo() {
+    type A/*goto implementation*/ interface {
+            Hello()
     }
 
-    func bar() {
-	foo();
+    type B struct{}
+
+    func (b *B) Hello() {}
+
+    type C struct{}
+
+    func (c *C) Hello() {}
+
+    func main() {
```

Comment:
Thanks to @ycycyyc for go code that have multiple implementations: https://github.com/yegappan/lsp/issues/171#issuecomment-1477453343